internal/xpkg/lint.go

```diff
@@ -88,6 +96,15 @@ func IsConfiguration(o runtime.Object) error {
 	return nil
 }
 
+// IsFunction checks that an object is a Function meta type.
+func IsFunction(o runtime.Object) error {
+	po, _ := TryConvert(o, &pkgmetav1alpha1.Function{})
+	if _, ok := po.(*pkgmetav1alpha1.Function); !ok {
```

Comment:
```suggestion
	if _, ok := o.(*pkgmetav1alpha1.Function); !ok {
```

I think we can skip conversion here and just test whether the supplied object is a `*pkgmetav1alpha1.Function`. `IsPackage` and `IsConfiguration` need to handle the fact that multiple versions of the same type (i.e. `v1` and `v1alpha1`) exist, but that's not yet the case here. I believe this should allow us to get rid of the conversion code until it's needed.

Reply:
Yup, that's what I was thinking.